packages/framework-integration-tests/integration/provider-unaware/end-to-end/delete.events.integration.ts

```diff
@@ -0,0 +1,485 @@
+    return resultId === null
+  })
+  return result.data.SpecialReportsReadModel
+}
+
+async function deleteEvent(
+  client: ApolloClient<unknown>,
+  entityId: string,
+  entityTypeName: string,
+  createdAt: string
+): Promise<boolean> {
+  const result = await client.mutate({
+    variables: {
+      entityId: entityId,
+      entityTypeName: entityTypeName,
+      createdAt: createdAt,
+    },
+    mutation: gql`
+      mutation HardDelete($entityId: String!, $entityTypeName: String!, $createdAt: String!) {
+        HardDelete(input: { entityId: $entityId, entityTypeName: $entityTypeName, createdAt: $createdAt })
+      }
+    `,
+  })
+  return result?.data.HardDelete
+}
+
+function queryByEntity(
+  client: ApolloClient<unknown>,
+  entity: string,
+  timeFilters?: EventTimeParameterFilter,
+  entityID?: string,
+  limit?: number
+): Promise<ApolloQueryResult<any>> {
+  const queryTimeFilters = timeFilters ? `, from:"${timeFilters.from}" to:"${timeFilters.to}"` : ''
+  const queryEntityID = entityID ? `, entityID:"${entityID}"` : ''
+  const queryLimit = limit ? `, limit:${limit}` : ''
+  return client.query({
+    query: gql`
+      query {
+        eventsByEntity(entity: ${entity}${queryEntityID}${queryTimeFilters}${queryLimit}) {
+            createdAt
+            entity
+            entityID
+            requestID
+            type
+            user {
+                id
+                roles
+                username
+            }
+            value
+            deletedAt
+        }
+      }
+    `,
+  })
+}
+
+async function queryProductByEntity(client: ApolloClient<unknown>, entityID: string): Promise<any> {
+  const result = await waitForIt(
+    () => queryByEntity(client, 'Product', undefined, entityID),
+    (result) => {
+      const events: Array<EventSearchResponse> = result.data['eventsByEntity']
+      const found = events?.find((event) => event.entityID === entityID)
+      return found !== undefined
+    }
+  )
+  return result.data['eventsByEntity']
+}
+
+async function queryCartByEntity(client: ApolloClient<unknown>, entityID: string): Promise<any> {
+  const result = await waitForIt(
+    () => queryByEntity(client, 'Cart', undefined, entityID),
+    (result) => {
+      const events: Array<EventSearchResponse> = result.data['eventsByEntity']
+      const found = events?.find((event) => event.entityID === entityID)
+      return found !== undefined
+    }
+  )
+  return result.data['eventsByEntity']
+}
+
+async function findEntities(client: ApolloClient<unknown>, entityName: string): Promise<PaginatedEntitiesIdsResult> {
+  const result = await client.mutate({
+    variables: {
+      entityName: entityName,
+      limit: 99999,
+    },
+    mutation: gql`
+      mutation EntitiesIdsFinder($entityName: String!, $limit: Float!) {
+        EntitiesIdsFinder(input: { entityName: $entityName, limit: $limit })
+      }
+    `,
+  })
+  return result?.data?.EntitiesIdsFinder
+}
```

Comment:
I guess that many of these methods are duplicated elsewhere or can be reused in other tests, and we could create a file with these reusable helpers for all the tests. Nonetheless, I guess this refactor should go in another PR

Reply:
Absolutely. Some other tests call _similar_ helper methods, but as you say, the work required to create that method probably falls outside this PR's scope.